src/client/ua_client_internal.h

```diff
@@ -177,6 +177,12 @@ struct UA_Client {
 #if UA_MULTITHREADING >= 100
     UA_Lock clientMutex;
 #endif
+
+    UA_Boolean isReverseConnect;
```

Comment:
I think we should keep all the connection state information in one place: in the SecureChannel.
It is easier to reason about a single state-machine than many coupled state machines.
Also we notify state changes to the user application in a callback that has the SecureChannel state.
And in the SecureChannel you already have the the ConnectionManager+uintptr-id combination.

The following would be similar to the existing logic and do all in one state-machine:

1) Add two additional states to the `UA_SecureChannelState` enum. Something like
- `UA_SECURECHANNELSTATE_REVERSE_LISTENING` and
- `UA_SECURECHANNELSTATE_REVERSE_CONNECTED` (before the RHE is received)

2) Store the "listening socket" in the SecureChannel when the listening state is set.

3) Once the server does a reverse connect you get called in the "connectionCallback" set for the listen connection.
From within this callback
- shutdown the listening socket (if required set a new context pointer for the current connection, as the connectionCallback is called again when the listening socket has fully closed)
- Replace the socket stored in the client's SecureChannel with the new "active connection"
- Set the state to `UA_SECURECHANNELSTATE_REVERSE_CONNECTED` or similar.